lib/assignments/bloc.dart

```diff
@@ -0,0 +1,95 @@
+import 'dart:convert';
+
+import 'package:flutter_cached/flutter_cached.dart';
+import 'package:flutter/foundation.dart';
+import 'package:schulcloud/app/app.dart';
+import 'package:schulcloud/courses/courses.dart';
+import 'package:schulcloud/hive.dart';
+
+import 'data.dart';
+
+class Bloc {
+  CacheController<List<Assignment>> assignments;
+  CacheController<List<Submission>> submissions;
+
+  Bloc({
+    @required StorageService storage,
+    @required NetworkService network,
+    @required UserFetcherService userFetcher,
+  })  : assert(storage != null),
+        assert(network != null),
+        assert(userFetcher != null),
+        assignments = HiveCacheController<Assignment>(
+          storage: storage,
+          parentKey: cacheAssignmentsKey,
+          fetcher: () async {
+            var response = await network.get('homework');
+            var body = json.decode(response.body);
+
+            return [
+              for (var data in body['data'] as List<dynamic>)
+                Assignment(
```

Comment:
It's still readable as-is and we asynchronously fetch the user inside the `Assignment` constructor, so we would have to do stuff with `Future.wait`, because a plain `List` doesn't support `asyncMap` for now... 😕